### ISSUE TITLE
data "pagerduty_service" - Still refreshing... until timeout/EOF

### ISSUE THREAD

**Issue by landro** [PagerDuty/terraform-provider-pagerduty#772]
### Terraform Version

Terraform v1.6.2 

### Affected Resource(s)
pagerduty_service

### Terraform Configuration Files
```hcl
locals {
  services = [
    "cloudteam-extended-service",
    "cloudteam-simple-service",
    "cloudteam-service",
    "datamanagement-service",
    "identityprovider-service",
    "infrastructure-batch-service",
    "infrastructure-obm-test-service",
    "infrastructure-service"
  ]
}

data "pagerduty_service" "default" {
  for_each = toset(local.services)
  name     = each.value
}
```

### Debug Output

```
2023-11-20T14:25:15.422Z [TRACE] cli: {"@level":"info","@message":"data.pagerduty_service.default[\"cloudteam-service\"]: Refreshing...","@module":"terraform.ui","@timestamp":"2023-11-20T14:25:13.935576Z","hook":{"resource":{"addr":"data.pagerduty_service.default[\"cloudteam-service\"]","module":"","resource":"data.pagerduty_service.default[\"cloudteam-service\"]","implied_provider":"pagerduty","resource_type":"pagerduty_service","resource_name":"default","resource_key":"cloudteam-service"},"action":"read"},"type":"apply_start"}

2023-11-20T14:25:24.478Z [TRACE] cli: {"@level":"info","@message":"data.pagerduty_service.default[\"cloudteam-service\"]: Still refreshing... [10s elapsed]","@module":"terraform.ui","@timestamp":"2023-11-20T14:25:23.936165Z","hook":{"resource":{"addr":"data.pagerduty_service.default[\"cloudteam-service\"]","module":"","resource":"data.pagerduty_service.default[\"cloudteam-service\"]","implied_provider":"pagerduty","resource_type":"pagerduty_service","resource_name":"default","resource_key":"cloudteam-service"},"action":"read","elapsed_seconds":10},"type":"apply_progress"}

......

2023-11-20T14:38:25.019Z [TRACE] cli: {"@level":"info","@message":"data.pagerduty_service.default[\"cloudteam-service\"]: Still refreshing... [13m10s elapsed]","@module":"terraform.ui","@timestamp":"2023-11-20T14:38:23.985374Z","hook":{"resource":{"addr":"data.pagerduty_service.default[\"cloudteam-service\"]","module":"","resource":"data.pagerduty_service.default[\"cloudteam-service\"]","implied_provider":"pagerduty","resource_type":"pagerduty_service","resource_name":"default","resource_key":"cloudteam-service"},"action":"read","elapsed_seconds":790},"type":"apply_progress"}
```

### Expected Behavior

Should lookup data without failing

### Actual Behavior

When running this though terraform cloud, we regularly see an EOF from the API

```
Error: Get "https://api.pagerduty.com/services?query=cloudteam-service": EOF
with data.pagerduty_service.default["cloudteam-service"]
on pagerduty.tf line 9, in data "pagerduty_service" "default":
```
We have seen similar output for other service names as well, but generally speaking only one service from the list (_local.list_) per run will fail.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by erose96** [comment]
I am getting similar behavior when running `terraform plan` and the state fails to get refreshed

**Comment by imjaroiswebdev** [comment]
Addressed in #807 and patch available in TF provider **v3.5.2**.
